packages/context/src/inject.ts

```diff
@@ -368,12 +368,43 @@ export namespace inject {
   };
 }
 
+/**
+ * Assert the target type inspected from TypeScript for injection to be the
+ * expected type
```

Comment:
Please explain what happens when the check does not pass: the function throws a descriptive error.

Reply:
Even better, rename this function to `assertTargetType`, I think that's the usual naming conventions for functions that are throwing an error if a check was not passed.